File: proxy/tests/script_engine/header_delete.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local header_append = {
	request = function(flow)
		flow.request.headers:delete("X-Header1")
		flow.request.headers["X-Header2"] = nil
		flow.request.headers["X-Header3"] = nil -- TODO: there is no 3rd way to delete headers in lua
	end,
	response = function(flow)
		flow.response.headers:delete("X-Header1")
		flow.response.headers["X-Header2"] = nil
		flow.response.headers["X-Header3"] = nil -- TODO: there is no 3rd way to delete headers in lua
	end,
}
Extensions = { header_append }

File: proxy/tests/script_engine/header_delete.py
from roxy import Extension


class DeleteHeader(Extension):
    def request(self, flow):
        flow.request.headers.delete("X-Header1")
        flow.request.headers["X-Header2"] = None
        del flow.request.headers["X-header3"]

    def response(self, flow):
        flow.response.headers.delete("X-Header1")
        flow.response.headers["X-Header2"] = None
        del flow.response.headers["X-header3"]


Extensions = [DeleteHeader()]
